packages/terra-notification-dialog/src/NotificationDialog.jsx

```diff
@@ -0,0 +1,216 @@
+import IconAlert from 'terra-icon/lib/icon/IconAlert';
+import IconError from 'terra-icon/lib/icon/IconError';
+import IconWarning from 'terra-icon/lib/icon/IconWarning';
+import IconInformation from 'terra-icon/lib/icon/IconInformation';
+import IconSuccess from 'terra-icon/lib/icon/IconSuccess';
+import classNames from 'classnames/bind';
+import 'terra-base/lib/baseStyles';
+import styles from './NotificationDialog.scss';
+
+const cx = classNames.bind(styles);
+
+const variants = {
+  ALERT: 'alert',
+  ERROR: 'error',
+  WARNING: 'warning',
+  INFO: 'info',
+  SUCCESS: 'success',
+  CUSTOM: 'custom',
+};
+
+const propTypes = {
+  /**
+   * Header of the notification-dialog.
+   */
+  header: PropTypes.string,
+  /**
+   * Title of the notification-dialog.
+   */
+  title: PropTypes.string,
+  /**
+   * Message of the notification-dialog.
+   */
+  message: PropTypes.string,
+  /**
+   * The Action of the primary button.
+   */
+  primaryAction: PropTypes.shape({
+    text: PropTypes.string,
+    onClick: PropTypes.func,
+  }),
+  /**
+   * The Action of the secondary button.
+   */
+  secondaryAction: PropTypes.shape({
+    text: PropTypes.string,
+    onClick: PropTypes.func,
+  }),
+  /**
+   * The variant of notification to be rendered.
+   * Use one of alert, error, warning, info, success, custom.
+   */
+  variant: PropTypes.oneOf([
+    variants.ALERT,
+    variants.ERROR,
+    variants.WARNING,
+    variants.INFO,
+    variants.SUCCESS,
+    variants.CUSTOM,
+  ]),
+  /**
+   * The icon to be used for a notification of type custom. This will not be used for any other notification types.
+   */
+  customIcon: PropTypes.element,
+  /**
+   * Toggle to show notification-dialog or not.
+   */
+  isOpen: PropTypes.bool.isRequired,
+  /**
+   * Callback function indicating a close condition was met, should be combined with isOpen for state management.
+   */
+  onRequestClose: PropTypes.func.isRequired,
+  /**
+   * A callback function to let the containing component (e.g. modal) to regain focus.
+   */
+  releaseFocus: PropTypes.func,
+  /**
+   * A callback function to request focus from the containing component (e.g. modal).
+   */
+  requestFocus: PropTypes.func,
+};
+
+const defaultProps = {
+  title: null,
+  message: null,
+  variant: variants.CUSTOM,
+};
+
+const actionSection = (primaryAction, secondaryAction) => {
+  let actionButton = null;
+  let dismissButton = null;
+  if (!primaryAction && !secondaryAction) {
+    return null;
+  }
+  if (primaryAction) {
+    actionButton = <Button text={primaryAction.text} variant={Button.Opts.Variants.EMPHASIS} onClick={primaryAction.onClick} />;
+  }
+  if (secondaryAction) {
+    dismissButton = <Button text={secondaryAction.text} onClick={secondaryAction.onClick} />;
+  }
+
+  return <div className={cx('actions')}>{actionButton}{dismissButton}</div>;
+};
+
+const getIcon = (variant, customIcon = null) => {
+  switch (variant) {
+    case variants.ALERT:
+      return (<IconAlert />);
```

Comment:
Need to change icons from being imported from terra-icon to being included via CSS to be properly re-themeable. Status-View is similar with both specified icons for the variants and also allows for custom. 

Reference for how it is done in the JSX:
https://github.com/cerner/terra-core/blob/master/packages/terra-status-view/src/StatusView.jsx#L159-L174

And in the SCSS:
https://github.com/cerner/terra-core/blob/master/packages/terra-status-view/src/StatusView.scss#L18-L44